cloud.ganttproject.colloboque/src/main/kotlin/cloud/ganttproject/colloboque/DevServer.kt

```diff
@@ -138,10 +146,19 @@ class ColloboqueWebSocketServer(port: Int, private val colloboqueServer: Collobo
     }
 
     override fun onClose(code: WebSocketFrame.CloseCode?, reason: String?, initiatedByRemote: Boolean) {
+      for (webSockets in connectedClients.values) {
+        webSockets.remove(this)
+      }
       LOG.debug("WebSocket closed")
     }
 
     override fun onMessage(message: WebSocketFrame) {
+      if (message.textPayload.startsWith("PROJECTREFID")) {
```

Comment:
Hm, LISTEN sounds a bit better. I'm not sure that sending refid with heartbeat is good idea, because i can imagine situation where we can open another project between heartbeats(is it even possible, or i am wrong?). Renamed for now